bootcamp-api.API/Services/CalendarEventService.cs

```diff
@@ -0,0 +1,126 @@
+using System;
+using System.Collections.Generic;
+using System.ComponentModel.Design;
+using System.Drawing;
+using System.Linq;
+using System.Xml.Linq;
+using Domain;
+using bootcamp_api.Data;
+using bootcamp_api.Exceptions;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.AspNetCore.Mvc;
+using AutoMapper;
+
+namespace bootcamp_api.Services
+{
+    public class CalendarEventService : ICalendarEventService
+    {
+
+        private readonly PawssierContext _context;
+        private readonly IMapper _mapper;
+
+        public CalendarEventService(PawssierContext context, IMapper mapper)
+        {
+            _context = context;
+            _mapper = mapper;
+        }
+
+        public Dto.CalendarEvent[] Get(ApiVersion version, int user_id, int month = 0, int year = 0)
+        {
+            var calendarEvents = new CalendarEvent[0];
+
```

Comment:
Using toArray(), or ToAnything is what executes the LINQ against the DB
Before you do that, you can keep modifying the query and adding new Where clauses to it.
See this article about how to update your linq query multiple times before you execute it.
https://stackoverflow.com/questions/10884651/linq-adding-conditions-to-the-where-clause-conditionally